cbcontainers/state/common/volumes.go

```diff
@@ -0,0 +1,67 @@
+package common
+
+import coreV1 "k8s.io/api/core/v1"
+
+const (
+	DesiredTlsRootCAsVolumeName          = "root-cas"
+	DesiredTlsRootCAsVolumeMountReadOnly = true
+)
+
+func GetVolumeIndexForName(templatePodSpec *coreV1.PodSpec, name string) int {
+	for i, volume := range templatePodSpec.Volumes {
+		if volume.Name == name {
+			return i
+		}
+	}
+
+	volume := coreV1.Volume{Name: name}
+	templatePodSpec.Volumes = append(templatePodSpec.Volumes, volume)
```

Comment:
I don't like how the method name is `GetSomething` but in the same time it mutates its arguments. Can we come up with a better name for it? also - it seems that it does 2 things - get the index of a volume OR add a new volume with the same name.

maybe `GetOrCreateVolumeForName` and a good documentation should be enough

Reply:
Agreed, I'd prefer if it returns -1 when the volume is not found so the caller can decide what to do instead

I guess it also does not have to be exported ?

Reply:
it is used outside of the package. 

I am OK with the function doing 2 things, I just think that we should name it better so that it is clear what exactly does it do.

Reply:
wdyt about `EnsureAndGetVolumeIndexForName`?

Reply:
I like it. this is a long name and it describes the action pretty clearly.

a shorter one could be `EnsureVolumeIndexForName` and in the godoc we can write that the return value is actually the one we are ensuring in the first place.

I think both are better than the current one, so whatever you fancy more.